tla/consistency/TraceMultiNodeReads.tla

```diff
@@ -0,0 +1,137 @@
+    "RwTxResponse" :>  RwTxResponse @@
+    "TxStatusReceived" :> TxStatusReceived @@
+    "RoTxRequest" :> RoTxRequest @@
+    "RoTxResponse" :>  RoTxResponse
+
+ToStatus ==
+    "CommittedStatus" :> CommittedStatus @@
+    "InvalidStatus" :>  InvalidStatus
+
+\* Beware to only prime e.g. inbox in inbox'[rcv] and *not* also rcv, i.e.,
+ \* inbox[rcv]'.  rcv is defined in terms of TLCGet("level") that correctly
+ \* handles priming, which causes for rcv' to equal rcv of the next log line.
+IsEvent(e) ==
+    \* Equals FALSE if we get past the end of the log, causing model checking to stop.
+    /\ l \in 1..Len(JsonLog)
+    /\ logline.action = e
+    /\ l' = l + 1
+
+IsRwTxRequestAction ==
+    /\ IsEvent("RwTxRequestAction")
+    /\ RwTxRequestAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].tx = logline.tx
+
+IsRwTxExecuteAction ==
+    /\ IsEvent("RwTxExecuteAction")
+    /\ RwTxExecuteAction
+    /\ history'[Len(history')].tx = logline.tx
+
+IsRwTxResponseAction ==
+    /\ IsEvent("RwTxResponseAction")
+    /\ RwTxResponseAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].tx = logline.tx
+
+IsStatusCommittedResponseAction ==
+    /\ IsEvent("StatusCommittedResponseAction")
+    /\ StatusCommittedResponseAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].status = ToStatus[logline.status]
+
+IsRoTxRequestAction ==
+    /\ IsEvent("RoTxRequestAction")
+    /\ RoTxRequestAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].tx = logline.tx
+
+IsRoTxResponseAction ==
+    /\ IsEvent("RoTxResponseAction")
+    /\ RoTxResponseAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].tx = logline.tx
+
+IsStatusInvalidResponseAction ==
+    /\ IsEvent("StatusInvalidResponseAction")
+    /\ StatusInvalidResponseAction
+    /\ history'[Len(history')].type = ToTxType[logline.type]
+    /\ history'[Len(history')].status = ToStatus[logline.status]
+
+TraceNext ==
+    \/ IsRwTxRequestAction
+    \/ IsRwTxExecuteAction
+    \/ IsRwTxResponseAction
+    \/ IsStatusCommittedResponseAction
+    \/ IsRoTxRequestAction
+    \/ IsRoTxResponseAction
+    \/ IsStatusInvalidResponseAction
+
+TraceSpec ==
+    TraceInit /\ [][TraceNext]_<<l, vars>>
+
+-------------------------------------------------------------------------------------
+
+Termination ==
+    l = Len(JsonLog) => TLCSet("exit", TRUE)
+
+TraceView ==
+    \* A high-level state  s  can appear multiple times in a system trace.  Including the
+     \* current level in TLC's view ensures that TLC will not stop model checking when  s
+     \* appears the second time in the trace.  Put differently,  TraceView  causes TLC to
+     \* consider  s_i  and s_j  , where  i  and  j  are the positions of  s  in the trace,
+     \* to be different states.
+    <<vars, l>>
```

Comment:
By default, TLC calculates a state hash/fingerprint based on all variable.  Thus `TraceView` can be removed (also in the config).  It is likely a left-over from when `TLCGet("level") was refactored into an ordinary variable `l`.